### ISSUE TITLE
Crash in VulkanSwapChain::hasResized()

### ISSUE THREAD

**Issue by jeanlemotan** [google/filament#6829]
After updating to the latest filament, main branch, my app crashes in `VulkanSwapChain::hasResized()`:

```cpp
bool VulkanSwapChain::hasResized() const {
    if (surface != VK_NULL_HANDLE) {
	return false;
    }
    VkSurfaceCapabilitiesKHR caps;
    vkGetPhysicalDeviceSurfaceCapabilitiesKHR(mPhysicalDevice, surface, &caps); // <<< --- CRASH HERE, surface is VK_NULL_HANDLE
    VkExtent2D perceivedExtent = caps.currentExtent;
    // Create the low-level swap chain.
    if (caps.currentExtent.width == VULKAN_UNDEFINED_EXTENT ||
        caps.currentExtent.height == VULKAN_UNDEFINED_EXTENT) {
        perceivedExtent = mFallbackExtent;
    }
    return !equivalent(clientSize, perceivedExtent);
}
```

I'm running on Windows 11, AMD Radeon RX 6900 XT GPU latest drivers. Release but with validation layers enabled, and I'm getting this error:

```
VUID-vkGetPhysicalDeviceSurfaceCapabilitiesKHR-surface-parameter(ERROR / SPEC): msgNum: -566353962 - Validation Error: [ VUID-vkGetPhysicalDeviceSurfaceCapabilitiesKHR-surface-parameter ] Object 0: handle = 0x245447ef7d0, type = VK_OBJECT_TYPE_INSTANCE; | MessageID = 0xde3e1fd6 | Invalid VkSurfaceKHR Object 0x0. The Vulkan spec states: surface must be a valid VkSurfaceKHR handle (https://vulkan.lunarg.com/doc/view/1.3.243.0/windows/1.3-extensions/vkspec.html#VUID-vkGetPhysicalDeviceSurfaceCapabilitiesKHR-surface-parameter)
    Objects: 1
        [0] 0x245447ef7d0, type: 1, name: NULL
```

The condition in the first line of the function looks wrong to me - I imagine the `hasResized` makes sense only for swap chains associated with a native surface (so when surface != VK_NULL_HANDLE), while for headless surfaces it should return false, since they never resize by themselves.

If I invert the condition, the crash is gone but I don't get any rendering (blank screen, no per-frame validation issues).

**Comment by poweifeng** [comment]
Yes, the swapchain part is influx due to refactoring but hopefully this will be addressed in my next PR. Appreciate your patience.

**Comment by poweifeng** [comment]
Fixed in #6830
